include/votca/xtp/matrixfreeoperator.h

```diff
@@ -23,8 +23,26 @@
 namespace votca {
 namespace xtp {
 
-class MatrixFreeOperator : public Eigen::EigenBase<Eigen::MatrixXd> {
+class MatrixFreeOperator;
+}
+}  // namespace votca
+namespace Eigen {
+namespace internal {
+// MatrixReplacement looks-like a SparseMatrix, so let's inherits its traits:
+template <>
+struct traits<votca::xtp::MatrixFreeOperator>
+    : public Eigen::internal::traits<Eigen::MatrixXd> {};
+}  // namespace internal
+}  // namespace Eigen
+
+namespace votca {
+namespace xtp {
+
+class MatrixFreeOperator : public Eigen::EigenBase<MatrixFreeOperator> {
```

Comment:
The change is a bit above my pay grade I'm afraid. But the eigen web page agrees with you.

Reply:
It is called the Curiously recurring template pattern, basically it is a compile time inheritance MatrixFreeOperator can use the methods of Eigen::Base